html5ever/benches/html5ever.rs

```diff
@@ -0,0 +1,95 @@
+#[macro_use]
+extern crate criterion;
+extern crate html5ever;
+
+use std::fs;
+use std::path::PathBuf;
+
+use criterion::{Criterion, black_box, ParameterizedBenchmark};
+
+use html5ever::tokenizer::{BufferQueue, TokenSink, Token, Tokenizer, TokenizerOpts, TokenSinkResult};
+use html5ever::tendril::*;
+
+struct Sink;
+
+impl TokenSink for Sink {
+    type Handle = ();
+
+    fn process_token(&mut self, token: Token, _line_number: u64) -> TokenSinkResult<()> {
+        // Don't use the token, but make sure we don't get
+        // optimized out entirely.
+        black_box(token);
+        TokenSinkResult::Continue
+    }
+}
+
+impl Sink {
+    fn run(input: Vec<StrTendril>, opts: TokenizerOpts) {
+        let mut tok = Tokenizer::new(Sink, opts.clone());
+        let mut buffer = BufferQueue::new();
+        for buf in input.into_iter() {
+            buffer.push_back(buf);
+            let _ = tok.feed(&mut buffer);
+        }
+        let _ = tok.feed(&mut buffer);
+        tok.end();
+    }
+}
+
+fn run_bench(c: &mut Criterion, name: &str, opts: TokenizerOpts) {
```

Comment:
It seems odd to take a opts argument and then every caller passes in Default::default.

Reply:
I copied the old code, I thought to leave the option in case someone will need it, in future. I guess YAGNI applies in this case.